coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -235,6 +246,64 @@ private SqlNode visitLateralViewInternal(ASTNode node, ParseContext ctx, boolean
     return joinNode;
   }
 
+  private SqlNode visitLateralViewJsonTuple(List<SqlNode> sqlNodes, List<SqlNode> aliasOperands, SqlCall sqlCall) {
+    /*
+     Represent
+        LATERAL VIEW json_tuple(json, p1, p2) jt AS a, b
+     as
+        LATERAL (
+          SELECT
+            IF(p1 is supported JSON key, get_json_object(json, '$["${p1}"]'), NULL) a,
+            IF(p2 is supported JSON key, get_json_object(json, '$["${p1}"]'), NULL) b
+        ) AS jt(a, b)
```

Comment:
According to the [Hive language manual](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+UDF#LanguageManualUDF-json_tuple), using `json_tuple` is more efficient than using `get_json_object`. Doing this rewrite on the Hive side seems that every downstream engine is affected by this rewrite. Do you think we can rather do it on the Presto side?

Reply:
i agree it would be better. I tried to model this that way, so that initially it's kept as json_tuple and unpacked only later, but i didn't find a way to do this.